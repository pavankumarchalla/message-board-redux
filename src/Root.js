import React, { Component } from 'react'
import {createStore, combineReducers, applyMiddleware, compose} from 'redux';
import {Provider} from 'react-redux';
import ReduxPromise from 'redux-promise';
import ReduxThunk from 'redux-thunk';
import { BrowserRouter } from "react-router-dom";
import {postReducer} from './reduces/post_reducer';
import {messageReducer} from './reduces/message_reducer';

const rootReducer = combineReducers ({
    posts: postReducer,
    messages: messageReducer
  });
  
  const logger = store => {
    return (next) => {
      return (action) => {
  
        const prevState = store.getState();
        const nextState = next(action);
        console.log("[Prev State]", prevState);
        console.log("[next State]", nextState);
        return nextState;
      };
    };
  };
  
  const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
  const store = createStore(rootReducer, {}, composeEnhancers(applyMiddleware(logger, ReduxPromise, ReduxThunk)));
class Root extends Component {
    render() {
        return (
            <Provider store={store}>
                <BrowserRouter>{this.props.children}</BrowserRouter>
            </Provider>
        )
    }
}

export default Root;
