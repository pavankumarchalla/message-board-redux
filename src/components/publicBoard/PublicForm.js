import React, { Component } from "react";
import {connect} from 'react-redux';
import * as actions from '../actions/action' 

class PublicForm extends Component {
  state = {
    inputText: ""
  };

  handleChange = event => {
    this.setState({
      inputText: event.target.value
    });
  };

  handleSubmit = event => {
    event.preventDefault();
    this.props.addPost(this.state.inputText);
    this.setState({
      inputText: ""
    });
  };

  handlePosts = event => {
    this.props.getPosts();
  }

  render() {
    return (
      <div>
        <textarea
          onChange={this.handleChange}
          value={this.state.inputText}
          className="form-control"
        />
        <button onClick={this.handleSubmit} className="btn btn-primary">
          Post
        </button>
        <button onClick={this.handlePosts} className="btn btn-success">
          GET POSTS
        </button>
      </div>
    );
  }
}

export default connect(null, actions)(PublicForm);
