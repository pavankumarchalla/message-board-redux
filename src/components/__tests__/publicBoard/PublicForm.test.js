import React from "react";
import { mount } from "enzyme";
import Root from "../../../Root";
import PublicForm from "../../publicBoard/PublicForm";

let wrapper;
beforeEach(() => {
  wrapper = mount(
    <Root>
      <PublicForm />
    </Root>
  );
});

afterEach(() => {
  wrapper.unmount();
});

describe("textarea on Change", () => {
  beforeEach(() => {
    wrapper.find("textarea").simulate("change", {
      target: { value: "Type something" }
    });
    wrapper.update();
  });

  it("should change value when user is typing in the text area", () => {
    expect(wrapper.find("textarea").prop("value")).toEqual("Type something");
  });

  it("should empty text area on submit", () => {
    wrapper.find("button.btn-primary").simulate("click");
    wrapper.update();
    expect(wrapper.find("textarea").prop("value")).toBe("");
  });
});

it("should render a text area", () => {
  expect(wrapper.find("textarea").length).toEqual(1);
});

it("should render a blue button", () => {
  expect(wrapper.find("button.btn-primary").length).toEqual(1);
});