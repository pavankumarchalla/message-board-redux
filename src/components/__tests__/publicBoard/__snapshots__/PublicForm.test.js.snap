// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check snapshot for publicForm 1`] = `
<div>
  <textarea
    className="form-control"
    onChange={[Function]}
    value=""
  />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Post
  </button>
  <button
    className="btn btn-success"
    onClick={[Function]}
  >
    GET POSTS
  </button>
</div>
`;
